OrbitGl/TrackManager.cpp

```diff
@@ -0,0 +1,449 @@
+  filter_ = absl::AsciiStrToLower(filter);
+  UpdateFilteredTrackList();
+}
+
+void TrackManager::UpdateFilteredTrackList() {
+  if (filter_.empty()) {
+    sorted_filtered_tracks_ = sorted_tracks_;
+    return;
+  }
+
+  sorted_filtered_tracks_.clear();
+  std::vector<std::string> filters = absl::StrSplit(filter_, ' ', absl::SkipWhitespace());
+  for (const auto& track : sorted_tracks_) {
+    std::string lower_case_label = absl::AsciiStrToLower(track->GetLabel());
+    for (auto& filter : filters) {
+      if (absl::StrContains(lower_case_label, filter)) {
+        sorted_filtered_tracks_.emplace_back(track);
+        break;
+      }
+    }
+  }
+}
+
+std::vector<int32_t>
+TrackManager::GetSortedThreadIds() {  // Show threads with instrumented functions first
+  std::vector<int32_t> sorted_thread_ids;
+  std::vector<std::pair<int32_t, uint32_t>> sorted_threads =
+      OrbitUtils::ReverseValueSort(thread_count_map_);
+
+  for (auto& pair : sorted_threads) {
+    // Track "kAllThreadsFakeTid" holds all target process sampling info, it is handled
+    // separately.
+    if (pair.first != orbit_base::kAllProcessThreadsTid) {
+      sorted_thread_ids.push_back(pair.first);
+    }
+  }
+
+  // Then show threads sorted by number of events
+  std::vector<std::pair<int32_t, uint32_t>> sorted_by_events =
+      OrbitUtils::ReverseValueSort(event_count_);
+  for (auto& pair : sorted_by_events) {
+    // Track "kAllThreadsFakeTid" holds all target process sampling info, it is handled
+    // separately.
+    if (pair.first == orbit_base::kAllProcessThreadsTid) continue;
+    if (thread_count_map_.find(pair.first) == thread_count_map_.end()) {
+      sorted_thread_ids.push_back(pair.first);
+    }
+  }
+
+  return sorted_thread_ids;
+}
+
+void TrackManager::UpdateMovingTrackSorting() {
+  // This updates the position of the currently moving track in both the sorted_tracks_
+  // and the sorted_filtered_tracks_ array. The moving track is inserted after the first track
+  // with a value of top + height smaller than the current mouse position.
+  // Only drawn (i.e. not filtered out) tracks are taken into account to determine the
+  // insertion position, but both arrays are updated accordingly.
+  //
+  // Note: We do an O(n) search for the correct position in the sorted_tracks_ array which
+  // could be optimized, but this is not worth the effort for the limited number of tracks.
+
+  int moving_track_previous_position = FindMovingTrackIndex();
+
+  if (moving_track_previous_position != -1) {
+    auto moving_track = sorted_filtered_tracks_[moving_track_previous_position];
+    sorted_filtered_tracks_.erase(sorted_filtered_tracks_.begin() + moving_track_previous_position);
+
+    int moving_track_current_position = -1;
+    for (auto track_it = sorted_filtered_tracks_.begin(); track_it != sorted_filtered_tracks_.end();
+         ++track_it) {
+      if (moving_track->GetPos()[1] >= (*track_it)->GetPos()[1]) {
```

Comment:
Again, please avoid `[]`.

Reply:
In this case GetPos() returns a Vec2. You still didn't like the []?

Reply:
Oh, then it is probably fine.